app/tests/test_config.py

```diff
@@ -123,3 +166,52 @@ def test_aws_secrets_manager_config_initialized(monkeypatch):
     assert config.FLASKS3_ACTIVE is False
     assert config.FLASKS3_CDN_DOMAIN == "test_flasks3_cdn_domain"
     assert config.FLASKS3_BUCKET_NAME == "test_flasks3_bucket_name"
+
+
+@mock_aws
+def test_aws_secrets_manager_config_variable_not_set_error(monkeypatch):
+    """
+    GIVEN AWS secret with secret_id `AWS_SM_CONFIG_SECRET_ID` is set
+    and a variable 'DEFAULT_DATE_FORMAT' is not set
+    WHEN Config is initialized
+    THEN it should raise an exception with error
+    """
+    secret_value = json.dumps(
+        {
+            "AWS_REGION": "test_aws_region",
+            "KEYCLOAK_BASE_URI": "test_keycloak_base_uri",
+            "KEYCLOAK_CLIENT_ID": "test_keycloak_client_id",
+            "KEYCLOAK_REALM_NAME": "test_keycloack_realm_name",
+            "KEYCLOAK_CLIENT_SECRET": "test_keycloak_client_secret",  # pragma: allowlist secret
+            "RECORD_BUCKET_NAME": "test_record_bucket_name",
+            "FLASKS3_ACTIVE": "False",
+            "FLASKS3_CDN_DOMAIN": "test_flasks3_cdn_domain",
+            "FLASKS3_BUCKET_NAME": "test_flasks3_bucket_name",
+            # "DEFAULT_DATE_FORMAT": "test_default_date_format",
+            "SECRET_KEY": "test_secret_key",  # pragma: allowlist secret
+            "DB_PORT": "5432",
+            "DB_HOST": "test_db_host",
+            "DB_USER": "test_db_user",
+            "DB_PASSWORD": "test_db_password",  # pragma: allowlist secret
+            "DB_NAME": "test_db_name",
+            "DEFAULT_PAGE_SIZE": "test_default_page_size",
+        }
+    )
+
+    ssm_client = boto3.client("secretsmanager")
+
+    ssm_client.create_secret(
+        Name="test_secret_id",
+        SecretString=secret_value,
+    )
+
+    monkeypatch.setenv(
+        "AWS_SM_CONFIG_SECRET_ID", "test_secret_id"
+    )  # pragma: allowlist secret
+
+    config = AWSSecretsManagerConfig()
+
+    with pytest.raises(KeyError) as error:
+        inspect.getmembers(config)
+
+    assert str(error.value) == "'DEFAULT_DATE_FORMAT'"
```

Comment:
should this be the error message?

Reply:
for consistency checking the key for both .env and aws config values